In [20]:
# Before moving, let's load the dataset
import pandas as pd
from scipy import stats
df = pd.read_csv("population_by_country_dataset.csv")

In [21]:
# All the population data
population = df['Population (2020)']

In [32]:
len(population)

235

In this notebook, we will self-study(I will be teaching myself :3), about the descriptive statistics. Firstly, we will talk about theory, then with Python code we make our hands dirty by using theory on data. For data, we won't use any theorical data instead we will use real-world data so that we can understand what is happening, ask question as well as will try to understand the data while practicing                

[Dataset](https://www.kaggle.com/tanuprabhu/population-by-country-2020) | We will use population by country dataset for this learning process. We have choosed population dataset as all the data is in numeric format, we will require no pre-processing, overall a very clean dataset for learning stats

# Descriptive Statistics


Before start learning topics, so what is descriptive statistics?
So we have dataset. Using descriptive stats, we can get idea of summary, describe the basic feature of the dataset. It's like to first fundamental to learn about any dataset. Note: Dataset should be numeric.

Example: let's say you have data for your salary for every month in a year. To have a good idea about how much you earned in a year, you can all month's salary and divide by 12(12 month=1 year). Then you will get a good understanding, how average you have income in a year. See, you got understanding of the data but not any prediction, conclusion. 

Types of Descriptive Stats:
- Destribution
- Central Tendency
- Variability or Dispersion

# Central Tendency

## Mean, Median, Mode

When we have a raw dataset, the best and great way to idea about the dataset to find average. By measuring it, we can see the centre of the dataset.


- Mean -> add up all the numbers and divide by how many numbers there are. `SUM(Numbers)/COUNT(Numbers)`
- Median ->Is the middle number. It is found by putting the numbers in orderand taking the actual middle number if there is one,or the average of the two middle numbers if not.
- Mode -> Is the most commonly occurring number

Let's see Mean, Median and Mode of the population data

In [7]:
population.mean()

33227444.280851062

In [8]:
population.median()

5460109.0

In [31]:
stats.mode([population.values])

ModeResult(mode=array([[1440297825, 1382345085,  331341050,  274021604,  221612785,
         212821986,  206984347,  164972348,  145945524,  129166028,
         126407422,  115434444,  109830324,  102659126,   97490013,
          90003954,   84495243,   84176929,   83830972,   69830779,
          67948282,   65298930,   60446035,   60012400,   59436725,
          54473253,   53968739,   51276977,   50976248,   46757980,
          45974931,   45267449,   43984569,   44019263,   43686577,
          40372771,   39074280,   37839255,   37799407,   36985624,
          34905942,   33551824,   33050211,   33032075,   32436963,
          31398811,   31181428,   29935468,   29225196,   28421581,
          27808395,   26655083,   26486282,   25798588,   25550683,
          24346468,   23824369,   21428970,   20997293,   20346106,
          19214608,   19211425,   19144605,   18815231,   18468257,
          17971382,   17688599,   17571053,   17141544,   16816539,
          16758448,   16502877, 

So, Mean, Median and Mode are both important, but we have to choose after checking what type of variable we have given. For example, you see Mode? We find that, every number occurs 1(As population per country), but it doesn't give us any information to find about data.

But if see median, it gives us a very good information about the data. We can see average population in all the countries is around 33 million where highest is around 212 million and lowest is 801. It gives us central of all the population of the countries.

But what does the median told us? You see median is around 5.4 million. It's way low than average. But what can it tell us for this dataset? You see median calculates at by arranging and taking middle amount. So, by comparing with median and mean, we can see there more data in the dataset with higher population. So, we have 235 countries population data, and most of the countries have very high population.

You see? We analyzed a good amount of information with only mean and median measurement. Great!

So, with comparing Mean Vs Median, we can detect extreme values or Outliers immidiately.